tests/zquantum/core/wip/circuits/conversions/qiskit_conversions_test.py

```diff
@@ -79,6 +79,29 @@ def test_matrices_are_equal(self, zquantum_gate, qiskit_gate):
         np.testing.assert_allclose(zquantum_matrix, qiskit_matrix)
 
 
+class TestU3GateConversion:
+    @pytest.mark.parametrize(
+        "theta, phi, lambda_",
+        [
+            (0, 0, 0),
+            (0, np.pi / 5, 0),
+            (np.pi / 3, 0, 0),
+            (0, 0, np.pi / 7),
+            (42, -20, 30),
+        ],
+    )
+    def test_matrices_are_equal_up_to_phase_factor(self, theta, phi, lambda_):
+        zquantum_matrix = np.array(
+            _builtin_gates.U3(theta, phi, lambda_).matrix
+        ).astype(np.complex128)
+        qiskit_matrix = qiskit.extensions.U3Gate(theta, phi, lambda_).to_matrix()
+
+        quotient = zquantum_matrix @ np.linalg.inv(qiskit_matrix)
+        target_quotient = np.diag(quotient).mean() * np.eye(2, dtype=np.complex128)
```

Comment:
WDYT about extracting the predicate into sth like `_is_identity_with_multiplier`?